pkg/tsdb/prometheus/querydata/request.go

```diff
@@ -84,19 +80,18 @@ func New(
 		ID:                 settings.ID,
 		URL:                settings.URL,
 		exemplarSampler:    exemplarSampler,
-		enableDataplane:    features.IsEnabledGlobally(featuremgmt.FlagPrometheusDataplane),
-		enableScope:        features.IsEnabledGlobally(featuremgmt.FlagPromQLScope),
 	}, nil
 }
 
 func (s *QueryData) Execute(ctx context.Context, req *backend.QueryDataRequest) (*backend.QueryDataResponse, error) {
+	cfg := backend.GrafanaConfigFromContext(ctx)
 	fromAlert := req.Headers["FromAlert"] == "true"
 	result := backend.QueryDataResponse{
 		Responses: backend.Responses{},
 	}
 
 	for _, q := range req.Queries {
-		query, err := models.Parse(q, s.TimeInterval, s.intervalCalculator, fromAlert, s.enableScope)
+		query, err := models.Parse(q, s.TimeInterval, s.intervalCalculator, fromAlert, cfg.FeatureToggles().IsEnabled("promQLScope"))
```

Comment:
Nit: `FeatureToggles()` performs a `strings.Split` and other slice operations under the hood so it can be expensive if repeated for each query ([see here](https://github.com/grafana/grafana-plugin-sdk-go/blob/225dd2c715b25ae14f04f76a73317b2a6c039ed6/backend/config.go#L47-L62)). I would extract this into its own variable outside the loop:

```go
hasPromQLScopeFeatureFlag := cfg.FeatureToggles().IsEnabled("promQLScope")
```